api/provider.go

```diff
@@ -0,0 +1,190 @@
+package api
```

Comment:
I was thinking that this might deserve being in a package under api for clarity. Especially as we add more. 
```
api/provider/
  - provider.go
  - github.go
  - bitbucket.go
  - gitlab.go
```

it doesn't seem like it really _needs_ to be part of the API directly. Not a requirement